geode-core/src/main/java/org/apache/geode/internal/cache/eviction/LRUListWithAsyncSorting.java

```diff
@@ -212,7 +215,8 @@ private boolean scanInProgress() {
   }
 
   private boolean hasThresholdBeenMet(int recentlyUsedCount) {
-    return size() > 0 && (double) recentlyUsedCount / size() >= this.scanThreshold;
+    return size() >= maxEvictionAttempts
```

Comment:
I don't understand why we compare size() to maxEvictionAttempts here. I think the old code had "size() > 0" just to prevent a divide by zero.
hasThresholdBeenMet is used to decide if we should kick off a background scan.
maxEvictionAttempts is used to terminate a foreground scan that is trying to find an entry that has not been recently used.
If you are intentionally trying to delay starting a scan until we have at least "maxEvictionAttempts" recentlyUsed entries then would it be better to say "recentlyUsedCount >= maxEvictionAttempts"?

Reply:
As far as fixing the unit test goes, couldn't you have fixed in by giving it a higher scanThreshold? It seems like this would make it harder to document when a background scan will start because two different attributes would be used instead of just one.

Reply:
The idea was that on small lists (i.e. those that have fewer entries in them than the max number we will foreground scan on attempting eviction) we should not be scanning asynchronously at all as there is little value in doing so, in addition to providing a fix for the tests. It has been discussed that we may wish to make this value configurable in the future, anyway. This could be a different (possibly hardcoded) value unrelated to how many entries we will scan when getting one for eviction, if that is thought to be clearer without unduly adding additional parameterization (would there be value in setting the two values differently?).

The scanThreshold is difficult to use to fix certain tests, because when both entries are recently used, even requiring 100% of entries by marked in the test will still result in a scan (we prevent invalid values like 120% from being set, currently).